src/modules/test/main.cpp

```diff
@@ -0,0 +1,136 @@
+            json_object_get_string(jsonTestRecipeMetadata, g_modulePath.c_str()),
+            json_object_get_string(jsonTestRecipeMetadata, g_mimPath.c_str()),
+            json_object_get_string(jsonTestRecipeMetadata, g_testRecipesPath.c_str()),
+        };
+
+        TestRecipes recipes = TestRecipeParser::ParseTestRecipe(recipeMetadata.m_testRecipesPath);
+        for (auto &recipe : *recipes)
+        {
+            recipe.m_metadata = recipeMetadata;
+            recipe.m_mimObjects = MimParser::ParseMim(recipeMetadata.m_mimPath);
+        }
+
+        testRecipes->insert(testRecipes->end(), recipes->begin(), recipes->end());
+    }
+
+    json_value_free(root_value);
+    return testRecipes;
+}
+
+TestRecipes LoadValuesFromCLI(char* modulePath, char* mimPath, char* testRecipesPath)
+{
+    TestRecipes testRecipes = std::make_shared<std::vector<TestRecipe>>();
+
+    TestRecipeMetadata recipeMetadata = {
+        modulePath,
+        mimPath,
+        testRecipesPath,
+    };
+
+    TestRecipes recipes = TestRecipeParser::ParseTestRecipe(recipeMetadata.m_testRecipesPath);
+    for (auto &recipe : *recipes)
+    {
+        recipe.m_metadata = recipeMetadata;
+        recipe.m_mimObjects = MimParser::ParseMim(recipeMetadata.m_mimPath);
+    }
+
+    testRecipes->insert(testRecipes->end(), recipes->begin(), recipes->end());
+
+    return testRecipes;
+}
+
+int main(int argc, char **argv)
+{
+    SetFullLogging(true);
+    testing::InitGoogleTest(&argc, argv);
+    TestRecipes testRecipes;
+    
+    if (argc == 2)
+    {
+        // Test definitions present - perform more extensive tests using Mim and test recipes
+        if (strcmp(".json", std::filesystem::path(argv[1]).extension().c_str()) == 0)
+        {
+            testRecipes = LoadValuesFromConfiguration();
+        }
+        else
+        {
+            // Perform basic module tests - MMI only
+            std::cout << "Performing basic module tests on '" << argv[1] << "'" << std::endl;
+            auto module = std::make_shared<ManagementModule>(argv[1]);
+
+            // See gtest.h for details on this test registration
+            // https://github.com/google/googletest/blob/v1.10.x/googletest/include/gtest/gtest.h#L2438
+            testing::RegisterTest(
+                "TestRecipes", "Basic-Module-Test", nullptr, "Basic-Module-Test", __FILE__, __LINE__,
+                    [module]()->RecipeFixture*
+                    { 
+                        return new BasicModuleTester(module); }
+                    );
+        }
+    }
+    else if (argc > 3)
+    {
+        // Use given module path, mim path, and test recipe path
+        if (std::filesystem::exists(argv[1]) && std::filesystem::exists(argv[2]) && std::filesystem::exists(argv[3]))
+        {
+            std::cout << "Module : " << argv[1] << std::endl;
+            std::cout << "Mmi    : " << argv[2] << std::endl;
+            std::cout << "Recipe : " << argv[3] << std::endl;
+
```

Comment:
You are using at least 3 different styles of logging to console (and/or to test log) in this code. Please pick one and use it consistently. 

As the code is using GTest, use Gtest tracing. Not direct writing to console (this code for example) or OSConfig logging.

Reply:
I have removed the OSConfig logging, but the test app uses the standard output stream, gtest itself doesn't provide the ability to just write diagnostic messages, it only writes to a provided output stream on assertion/expected failures.